src/NuGet.Core/NuGet.Commands/PackagesLockFileBuilder.cs

```diff
@@ -89,5 +94,56 @@ public PackagesLockFile CreateNuGetLockFile(LockFile assetsFile)
             return lockFile;
         }
 
+        public async Task<PackagesLockFile> CreateNuGetLockFileAsync(IEnumerable<PackageReference> installedPackages, Func<PackageIdentity, string> getNupkgFilePath, CancellationToken token)
+        {
+            Debug.Assert(installedPackages != null);
+            Debug.Assert(getNupkgFilePath != null);
+
+            var lockFile = new PackagesLockFile();
+
+            foreach (var targetFramework in installedPackages.GroupBy(p => p.TargetFramework).OrderBy(g => g.Key))
```

Comment:
what happens when there is no `targetFramework` on installed packages in `Packages.config`? How are you categorizing those?

Reply:
The class that reads packages.config sets the TargetFramework to `Unsupported,V0.0`. I can map this to `Any,V0.0` in the builder if Unsupported looks too ugly in the lock file.

Reply:
`Any` makes more sense to me.